CacheMultimap.adoc

```diff
@@ -0,0 +1,79 @@
+= Native cache Mutimap
+
+The goal is to provide Infinispan Native CacheMultimap support. The proposal here is to start with the smallest interface implementation, in order to provide a proper API to the first multimap client, vert-x project.
+However, this API will evolve, and this first implementation must work on Embedded and Server mode.
+This first Multimap won't support duplicate values on the same key. 
+
+[source, java]
+.CacheMultimap.java
+----
+public interface CacheMultimap<K, V> {
+   boolean put(K key, V value); // <1> <2>
+
+   Collection<V> get(K key); // <3>
+
+   boolean remove(K key); //<4>
```

Comment:
the problem is that if the API defines having a return type then we need to actually send back the return (ie. getAndPut). But there may be cases where the user only wants to put the value and doesn't care about the return value (ie. put). The JCache API defines put as being void for example.
We have a way to work around this despite inheriting from Map with our `Flag.SKIP_REMOTE_LOOKUP` but no one knows to use this.